platforms/solana/src/platform.ts

```diff
@@ -176,41 +178,74 @@ export class SolanaPlatform<N extends Network>
     stxns: SignedTx[],
     opts?: SendOptions,
   ): Promise<TxHash[]> {
-    const { blockhash, lastValidBlockHeight } = await this.latestBlock(rpc);
-    const txhashes = await Promise.all(
-      stxns.map((stxn) =>
-        rpc.sendRawTransaction(
-          stxn,
-          // Set the commitment level to match the rpc commitment level
-          // otherwise, it defaults to finalized
-          opts ?? { preflightCommitment: rpc.commitment },
-        ),
-      ),
-    );
-
     const results = await Promise.all(
-      txhashes.map((signature) => {
-        return rpc.confirmTransaction(
-          {
-            signature,
-            blockhash,
-            lastValidBlockHeight,
-          },
-          rpc.commitment,
-        );
-      }),
+      stxns.map((stxn) => this.sendTxWithRetry(rpc, stxn, opts)),
     );
 
+    const txhashes = results.map((r) => r.signature);
+
     const erroredTxs = results
-      .filter((result) => result.value.err)
-      .map((result) => result.value.err);
+      .filter((r) => r.response.value.err)
+      .map((r) => r.response.value.err);
 
     if (erroredTxs.length > 0)
       throw new Error(`Failed to confirm transaction: ${erroredTxs}`);
 
     return txhashes;
   }
 
+  static async sendTxWithRetry(
+    rpc: Connection,
+    tx: SignedTx,
+    sendOpts: SendOptions = {},
+    retryInterval = 5000,
+  ): Promise<{
+    signature: string;
+    response: RpcResponseAndContext<SignatureResult>;
+  }> {
+    const commitment = sendOpts.preflightCommitment ?? rpc.commitment;
+    const signature = await rpc.sendRawTransaction(tx, {
+      ...sendOpts,
+      skipPreflight: false, // The first send should not skip preflight to catch any errors
```

Comment:
what is the default? assuming false, can this be removed?

Reply:
Yeah it's false by default, but I would prefer to be explicit here with the accompanying comment